src/options.js

```diff
@@ -17,6 +17,8 @@ export function valueof(data, value, type) {
     ? map(data, constant(value), type)
     : typeof value?.transform === "function"
     ? maybeTypedArrayify(value.transform(data), type)
+    : value && data?.reindex
+    ? maybeTypedMap(data.reindex, (i) => value[i], type)
```

Comment:
This assumes that `value` is an array, but it’s technically possible to be an arbitrary iterator. (Though, I discovered there’s some confusion in isOptions.)